test/parallel/test-buffer-write-fast.js

```diff
@@ -0,0 +1,38 @@
+// Flags: --expose-internals --no-warnings --allow-natives-syntax
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+
+const { internalBinding } = require('internal/test/binding');
+
+function testFastUtf8Write() {
+  {
+    const buf = Buffer.from('\x80');
+
+    assert.strictEqual(buf[0], 194);
+    assert.strictEqual(buf[1], 128);
+  }
+
+  {
+    const buf = Buffer.alloc(64);
+    const newBuf = buf.subarray(0, buf.write('éñüçßÆ'));
+    assert.deepStrictEqual(newBuf, Buffer.from([195, 169, 195, 177, 195, 188, 195, 167, 195, 159, 195, 134]));
+  }
+
+  {
+    const buf = Buffer.alloc(64);
+    const newBuf = buf.subarray(0, buf.write('¿'));
+    assert.deepStrictEqual(newBuf, Buffer.from([194, 191]));
+  }
+}
+
+eval('%PrepareFunctionForOptimization(Buffer.prototype.utf8Write)');
+testFastUtf8Write();
+eval('%OptimizeFunctionOnNextCall(Buffer.prototype.utf8Write)');
+testFastUtf8Write();
+
+if (common.isDebug) {
+  const { getV8FastApiCallCount } = internalBinding('debug');
+  assert.strictEqual(getV8FastApiCallCount('buffer.writeString'), 3);
+}
```

Comment:
```suggestion
'use strict';

const { test, describe } = require('node:test');
const common = require('../common');

const { internalBinding } = require('internal/test/binding');

function testFastUtf8Write() {
  test('Single UTF-8 character', (t) => {
    const buf = Buffer.from('\x80');
    t.assert.strictEqual(buf[0], 194);
    t.assert.strictEqual(buf[1], 128);
  });

  test('Multiple UTF-8 characters', (t) => {
    const buf = Buffer.alloc(64);
    const newBuf = buf.subarray(0, buf.write('éñüçßÆ'));
    t.assert.deepStrictEqual(newBuf, Buffer.from([195, 169, 195, 177, 195, 188, 195, 167, 195, 159, 195, 134]));
  });

  test('Single UTF-8 character', (t) => {
    const buf = Buffer.alloc(64);
    const newBuf = buf.subarray(0, buf.write('¿'));
    t.assert.deepStrictEqual(newBuf, Buffer.from([194, 191]));
  });
}

eval('%PrepareFunctionForOptimization(Buffer.prototype.utf8Write)');
describe('%PrepareFunctionForOptimization(Buffer.prototype.utf8Write)', testFastUtf8Write);
eval('%OptimizeFunctionOnNextCall(Buffer.prototype.utf8Write)');
describe('%OptimizeFunctionOnNextCall(Buffer.prototype.utf8Write)', testFastUtf8Write);

if (common.isDebug) {
  const { getV8FastApiCallCount } = internalBinding('debug');
  test('Check V8 Fast API call count', (t) => {
    t.assert.strictEqual(getV8FastApiCallCount('buffer.writeString'), 3);
  });
}
```

Fair warning: I haven't tested this code. I just wrote it up in the Github UI, so please please test this before use.

Reply:
As far as I know we don't write node test with node:test...